src/Commands/TransactionChecker.php

```diff
@@ -0,0 +1,186 @@
+use Enjin\Platform\Services\Processor\Substrate\ExtrinsicProcessor;
+use Facades\Enjin\Platform\Services\Processor\Substrate\State;
+use Illuminate\Console\Command;
+use Illuminate\Support\Arr;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+class TransactionChecker extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    public $signature = 'platform:transaction-checker';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    public $description;
+
+    protected Codec $codec;
+    protected Substrate $client;
+    protected ProgressBar $progressBar;
+    protected Carbon $start;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->description = __('enjin-platform::commands.transactions.description');
+        $this->start = now();
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(Substrate $client, Codec $codec): void
+    {
+        $this->codec = $codec;
+        $this->client = $client;
+
+        $blockNumber = Block::where('synced', true)->max('number');
+        $maxBlockToCheck = $blockNumber - 100;
+
+        $transactions = collect(Transaction::where([
+            'state' => TransactionState::BROADCAST,
+        ])->whereNotNull(['signed_at_block', 'transaction_chain_hash'])
+            ->where('signed_at_block', '<', $maxBlockToCheck)->get());
+
+        if ($transactions->isEmpty()) {
+            $this->info('There are no transactions to check.');
+
+            return;
+        }
+
+        $minSignedAtBlock = $transactions->min('signed_at_block');
+        $this->info(__('enjin-platform::commands.transactions.header'));
+        $this->info(__('enjin-platform::commands.transactions.syncing', ['fromBlock' => $minSignedAtBlock, 'toBlock' => $maxBlockToCheck]));
+
+        if ($minSignedAtBlock > $maxBlockToCheck) {
+            $this->info('There are no transactions to check in those blocks');
+
+            return;
+        }
+
+        $this->info(__('enjin-platform::commands.transactions.fetching'));
+        $counter = $transactions->count();
+        $hashes = array_filter($transactions->pluck('transaction_chain_hash')->toArray());
+
+        $this->progressBar = $this->output->createProgressBar($counter);
+        $this->progressBar->setFormat('debug');
+        $this->progressBar->start();
+
+        for ($i = $minSignedAtBlock; $i <= $maxBlockToCheck; $i++) {
+            $this->progressBar->setProgress($counter - collect($transactions)->count());
+
+            $block = Block::firstWhere('number', $i);
+            if (!($block?->hash)) {
+                $block = Block::updateOrCreate(
+                    ['number' => $i],
+                    ['hash' => $client->callMethod('chain_getBlockHash', [$i])],
+                );
+            }
+
+            $extrinsics = $this->fetchExtrinsics($block, $client);
```

Comment:
On second thought, it might be better to create a new function in BlockProcessor and shift all the logic there, this way it will be easy for you to reuse existing functions we have there.